packages/material-ui-lab/src/Breadcrumb/Breadcrumb.js

```diff
@@ -0,0 +1,175 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { withStyles } from '@material-ui/core/styles';
+import { emphasize } from '@material-ui/core/styles/colorManipulator';
+import ButtonBase from '@material-ui/core/ButtonBase';
+
+const height = 24;
+
+const styles = theme => ({
+  button: {
+    fontFamily: theme.typography.fontFamily,
+    fontSize: theme.typography.fontSize,
+    height,
+    lineHeight: `${height}px`,
+    color: theme.palette.grey[800],
+    borderRadius: 2,
+    '&:hover, &:focus': {
+      backgroundColor: theme.palette.grey[100],
+    },
+    '&:active': {
+      boxShadow: theme.shadows[0],
+      backgroundColor: emphasize(theme.palette.grey[100], 0.12),
+    },
+  },
+  active: {
+    fontWeight: theme.typography.fontWeightMedium,
+  },
+  gutters: {
+    paddingLeft: 8,
+    paddingRight: 8,
+  },
+  icon: {
+    marginRight: 4,
+    width: height / 1.2,
+    height: height / 1.2,
+    color:
+      theme.palette.type === 'light'
+        ? theme.palette.grey[700]
+        : theme.palette.grey[300],
+  },
+});
+
+function DefaultComponent(props) {
+  const { className, label, icon, active, tabIndex, onClick, ...rest } = props;
+
+  return (
+    <ButtonBase
+      variant="default"
+      color="primary"
+      className={className}
+      tabIndex={tabIndex}
+      onClick={onClick}
+      disabled={active}
+      {...rest}
+    >
+      {icon}
+      {label}
+    </ButtonBase>
+  );
+}
+
+class Breadcrumb extends React.PureComponent {
+  render() {
+    const {
+      classes,
+      className,
+      component: Component,
+      icon: iconProp,
+      label,
+      active,
+      disableGutters,
+      tabIndex,
+      onClick,
+      ...other
+    } = this.props;
+
+    let newTabIndex = tabIndex;
+
+    if (!newTabIndex) {
+      newTabIndex = onClick ? 0 : -1;
+    }
+
+    let icon = null;
+    if (iconProp && React.isValidElement(iconProp)) {
+      icon = React.cloneElement(iconProp, {
+        className: classNames(classes.icon, iconProp.props.className),
+      });
+    }
+    return Component ? (
```

Comment:
> I see that the components currently receives different props but I don't understand why

It looks like it relates to the use of the `active` prop to control `disabled` on the button, and since Chip (used in the customization example) doesn't support `disabled` (perhaps it should, regardless of this PR?), the `active prop` isn't being passed.

Instead, it should be being pulled out in the customized component, and either "discarded", or used some other way - in the case of Chip, it could be used to toggle the `onClick` value - the current way to disable Chip.

Reply:
The code should somehow reflect that. If there are specific props set that are not HTML attributes we should also add that to the documentation somehow.

Reply:
Perhaps `component` isn't even the right name for this prop, since that has consistent usage in other components that have it, "The component used for the root node. Either a string to use a DOM element or a component.", whereas here it has to be a custom react component to handle the props passed down.

Actually, looking it, I think it might be best to remove the prop altogether, and simply use the custom BreadCrumb in the customization example as a direct replacement for breadcrumb.

That further simplifies the implementation, and removes the problem entirely.

Reply:
Sounds like a already fell into that trap. I assumed it was working like any other `component` prop in  other component APIs.